pkg/httputils/mimetype.go

```diff
@@ -0,0 +1,40 @@
+package httputils
+
+import (
+	"bufio"
+	"errors"
+	"mime"
+	"net/http"
+)
+
+var MimeTypes = struct {
+	TextPlain   string
+	Tar         string
+	OctetStream string
+}{"text/plain", "application/tar", "application/octet-stream"}
+
+// DetectContentType returns a best guess representation of the MIME
+// content type for the bytes available at r.  The value detected by
+// http.DetectContentType is guaranteed not be nil, defaulting to
+// application/octet-stream when a better guess cannot be made. The
+// result of this detection is then run through mime.ParseMediaType()
+// which separates it from any parameters.
+// Note that calling this function does not advance the Reader at r
+func DetectContentType(r *bufio.Reader, plen int) (string, map[string]string, error) {
+	if plen <= 0 {
```

Comment:
since this function is decoupled from the actual http response reading process I believe we should not work around a negative plen. Anyway, I'm thinking if it would be better to just pass a `[]byte` slice instead of a *Reader.